### PR TITLE
Terraform base script

### PR DESCRIPTION
**Pull request by gopal-kannan**
Working base script creating necessary components.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,10 +1 @@
...
-Repository for SAEB Analytics Platform configuration
-
-This repository will store the TeamCity and Terraform scripts to be able to deploy the SAEB Analytics Platform configurations into Azure with TeamCity.
-
-It will leverage the following repositories to bring the code into the Azure resources respectively that will be used to execute the "data pipelines"
-
-- [AP-Databricks](https://github.com/DTS-STN/AP-Databricks)
-- [AP-DataFactory](https://github.com/DTS-STN/AP-DataFactory)
-- AP-Synapse (to be created)
+# saeb
```
Comment (s-laugh) [inline README.md:1]: This file shouldn't get overridden like this.

File: main.tf [hunk 2]
```diff
@@ -0,0 +1,116 @@
+# Declare some local variables
+locals {
+  build_environment = var.environment == "production" ? "Prod" : "Dev"
+  common_tags = {
+    Product      = "SAEB Platform"
```
Comment (s-laugh) [inline main.tf]:
```suggestion
    ProductPortfolio = "SAEB"
    Product      = "SAEB Analytics Platform"
```

File: main.tf [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+  length = 4
+  upper = false
+  special = false
+  number = false
+}
+
+
+# Create storage account for main input and curated CSV files as well as for synapse filesystem
+resource "azurerm_storage_account" "saeb_storage" {
+  name                     = "stsaebdevca01${random_string.random.result}"
```
Comment (s-laugh) [inline main.tf]: We might have to loop back on our naming conventions unless we can include a dynamic environment component to the name.
```suggestion
  name                     = "stapsndbxca01${random_string.random.result}"
```

**Review by s-laugh — CHANGES_REQUESTED** [review]
Looks good! Just some minor tweaks.
We can loop back in another place about managing the naming conventions and ensuring the env part is injected accurately.

File: README.md (hunk 1 above)
Comment (gopal-kannan) [reply README.md:1]: Necessary content is added.

**Comment by gopal-kannan** [comment]
Please review and let me know your comments if any.

File: README.md (hunk 1 above)
Comment (s-laugh) [reply README.md:1]: I'm not really clear on the need for the changes in this file - it's now adding a list of files, but that can be found though the directory and this PR/commit history, the readme shouldn't need to identify them; and their is already a header.

**Review by s-laugh — COMMENTED** [review]
I'm not understanding the changes to the readme; also wanted to double check that all the names for resources were reviewed to match the latest in the naming conventions (i only saw the one change I suggested).

File: README.md (hunk 1 above)
Comment (gopal-kannan) [reply README.md:1]: This document should be living document and grows as we make changes. If more changes required, we can discuss it and determine what are all the content to go and instructions to the developers what to add as and when they commit new changes.

Comment (gopal-kannan) [reply README.md:1]: I added some content. Please let me know your feedback.

Comment (s-laugh) [reply README.md:1]: looks good

Comment (s-laugh) [reply README.md:1]: looks good

File: main.tf [hunk 4]
```diff
@@ -0,0 +1,117 @@
+# Declare some local variables
+locals {
+  build_environment = var.environment == "production" ? "Prod" : "Dev"
```
Comment (s-laugh) [inline main.tf]: glad we have a variable for this, but I don't see it being used anywhere and it's missing a "Sndbx" state - it should effect the name of each resource and be a common tag, the tag should have the full name "Sandbox", "Development", "Production".

Comment (gopal-kannan) [reply main.tf]: The environment variable configuration is complete. The github configuration for development is still pending due to the challenge we have with firewall / proxy.

**Comment by gopal-kannan** [comment]
This is the most recent version of tf code which can go to main branch. @s-laugh please approve the merge so we have a base version saved in main branch. Thanks.

**Review by s-laugh — APPROVED** [review]

**Comment by gopal-kannan** [comment]
@s-laugh  I am not authorized to merge this pull request to main branch. Could you please merge it? Thanks.

**Pull request merged**